third_problem.rb

```diff
@@ -0,0 +1,22 @@
+class ThirdProblem
+  def initialize(numbers_array)
+    @numbers = numbers_array
+  end
+
+  def hasDuplicates
+    found = false
+    @numbers.each {|index|
+      absolute_current_index =index.abs-1
+      absolut_target_index = @numbers[absolute_current_index].abs-1
+      if @numbers[absolut_target_index] < 0
+        found = true
+        break
+      end
+      @numbers[absolut_target_index] = -@numbers[absolut_target_index]
+    }
+    found
```

Comment:
If you where allowed to use a different data structure to solve this problem what data structure will you use and why?